polkadot/statement-table/src/generic.rs

```diff
@@ -387,9 +385,10 @@ impl<Ctx: Context> Table<Ctx> {
 		authority: Ctx::AuthorityId,
 		candidate: Ctx::Candidate,
 		signature: Ctx::Signature,
+		group: Ctx::GroupId,
 	) -> ImportResult<Ctx> {
-		let group = Ctx::candidate_group(&candidate);
```

Comment:
I think we should add a check that the `ParaId` assigned to the core matches the paraid that got signed in the statement. Otherwise validators could sign garbage (not assigned paraid candidates), knowing they will be filtered out in the runtime, but DoSing the parachain as the legitimate statements are no longer accepted.

Reply:
(better to do this right where we determine the core from the statement - see other comment)

Reply:
done